src/v/security/scram_authenticator.cc

```diff
@@ -34,6 +35,8 @@ scram_authenticator<T>::handle_client_first(bytes_view auth_bytes) {
     _principal = credential->principal().value_or(
       acl_principal{principal_type::user, authid});
     _credential = std::make_unique<scram_credential>(std::move(*credential));
+    _audit_user.name = _principal.name();
```

Comment:
This was done in case in case `authid` did not end up being the actual credentialed principal user name (I realize that probably never happens, they're always the same).

The purpose for this audit user was to ensure, in the case of authn failure, that, the audit log would contain the username of the user that failed to authenticate.

`authid` is an optional field.  For things like mTLS and kerberos that's filled in with either the DN or the kerberos principal mostly to ensure the auditor has a full context as to how the actual user was derived.

Reply:
I don't really know what `credential_uid` represents, but I'd be tempted to set it directly to `authid`, i.e., what SCRAM gives us, and before it's mapped (which can fail).

Reply:
`credential_uid` is defined as:

> The unique identifier of the user's credential. For example, AWS Access Key ID.

(source: https://schema.ocsf.io/1.0.0/objects/user?extensions=)

Reply:
> `credential_uid` is defined as:
> 
> > The unique identifier of the user's credential. For example, AWS Access Key ID.

Cool, that's why I suggest using the unmapped credential, in general.